src/widgets/DataTable/DataTableRow.js

```diff
@@ -42,187 +42,197 @@ import { formatStatus } from '../../utilities/index';
-                containerStyle={touchableCellContainer}
-                width={width}
-                isLastCell={isLastCell}
-              />
-            );
+          // This cell is disabled if the pageObject is finalised, the row has been explicitly set
+          // as disabled, or the rowData is disabled (i.e. data is an invoice),
+          const isDisabled =
+            isFinalised || (rowState && rowState.isDisabled) || rowData.isFinalised;
 
-          case 'editableDate':
-            return (
-              <NewExpiryDateInput
-                key={columnKey}
-                value={rowData[columnKey]}
-                rowKey={rowKey}
-                columnKey={columnKey}
-                editAction={getAction(columnKey)}
-                isFocused={isFocused}
-                isDisabled={isDisabled}
-                dispatch={dispatch}
-                width={width}
-                isLastCell={isLastCell}
-                rowIndex={rowIndex}
-              />
-            );
+          // Alignment of this particular column. Default to left hand ide.
+          const cellAlignment = alignText || 'left';
 
-          case 'status':
-            return (
-              <Cell
-                key={columnKey}
-                value={formatStatus(rowData[columnKey])}
-                width={width}
-                viewStyle={cellContainer[cellAlignment]}
-                textStyle={cellText[cellAlignment]}
-                isLastCell={isLastCell}
-              />
-            );
+          switch (type) {
+            case 'editableString':
+            case 'editableNumeric':
+              return (
+                <TextInputCell
+                  key={columnKey}
+                  value={rowData[columnKey]}
+                  rowKey={rowKey}
+                  columnKey={columnKey}
+                  editAction={getAction(columnKey)}
+                  isDisabled={isDisabled}
+                  dispatch={dispatch}
+                  width={width}
+                  viewStyle={cellContainer[cellAlignment]}
+                  textViewStyle={editableCellTextView}
+                  isLastCell={isLastCell}
+                  keyboardType={type === 'editableNumeric' ? 'numeric' : 'default'}
+                  textInputStyle={cellText[cellAlignment]}
+                  textStyle={editableCellUnfocused[cellAlignment]}
+                  cellTextStyle={editableCellText}
+                  rowIndex={rowIndex}
+                />
+              );
 
-          case 'date':
-            return (
-              <Cell
-                key={columnKey}
-                value={rowData[columnKey] && rowData[columnKey].toDateString()}
-                width={width}
-                viewStyle={cellContainer[cellAlignment]}
-                textStyle={cellText[cellAlignment]}
-                isLastCell={isLastCell}
-              />
-            );
+            case 'editableDate':
+              return (
+                <NewExpiryDateInput
```

Comment:
Everything up to this point is generic / flexible, which is why I'm querying: It is an *editableDate* type and the control is `NewExpiryDateInput`. Perhaps the control should be simply `DateInput`?

Reply:
mm - I suppose I was copying the previous component, but the component essentially allows for dates in the form `mm/yyyy` and will prepend the last of the previous month (apparently all expiries are the last of the month?)